### DISCUSSION TITLE
Icon button e link com state: disabled

### DISCUSSION THREAD

**Discussion by Daniela-Lorenzetti** [TiendaNube/nimbus-design-system#74] [Ideas] [answered]
Na nova funcionalidade que estamos criando em Core, Metafields, mostramos, dependendo do plano do merchant, uma página de upsell com os cards e componentes desabilitados.

Como nesses casos, talvez seja útil em outros momentos ter essa versão de icon button e link em state disabled.

[Exemplo link desabilitado](https://www.figma.com/file/BmrBclFBUqDMkvN2D6PGY9/%F0%9F%AA%84-Metafields---Variantes-de-Produto?node-id=3775%3A516547&t=RvQIAzfsldwK7bUR-1)

[Exemplo de icon button desabilitado](https://www.figma.com/file/BmrBclFBUqDMkvN2D6PGY9/%F0%9F%AA%84-Metafields---Variantes-de-Produto?node-id=3775%3A516563&t=RvQIAzfsldwK7bUR-1)

**Comment by jcampos89** [comment, marked as answer]
Obrigado pela sugestão @Daniela-Lorenzetti ! 

**Sobre ter o componente link com disabled:**
Quando trabalhamos no refactor desse compontente cogitamos essa possibilitade, porém apesar do Nimbus ser feito em react todos os componentes utilizam a mesma lógica do Html herdando seus respectivos estados, nesse caso o link não possui status  disabled para manter uma consistência com os estados do Html, por isso optamos por não criar um status específico dentro do  Nimbus.

**Sobre ter o Icon Button com disabled**
Analisamos o componente e por ele ser um botão o HTML permite ter um status disabled, porém ainda não agregamos esse estilo ao componente. Olhando para o exemplo que você compartilhou, entendemos que o uso desse status disabled no kebab não faria tanto sentido, afinal ele poderia ser confundido com um botão ativo, já que não existem tantas referências nesse caso de uso.

Pensando nisso recomendamos que nesse caso o IconButton seja somente ocultado, não utilizando o status de disabled. 
De qualquer forma vamos agregar o estilo ao status para usos futuros, porém ainda precisamos entender no detalhe quais seriam suas regras de utilização.

Qualquer dúvida estou a disposição.
